snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/validation/Rf2GlobalValidator.java

```diff
@@ -124,42 +144,86 @@ public void validateTerminologyComponents(
 					}
 				});
 			
+			skippableMemberDependenciesByEffectiveTime.keySet().stream()
+				.forEach(id -> {
+					switch (SnomedIdentifiers.getComponentCategory(id)) {
+						case CONCEPT: conceptIds.add(id);
+							break;
+						case DESCRIPTION: descriptionIds.add(id);
+							break;
+						case RELATIONSHIP: relationshipIds.add(id);
+							break;
+						default: log.error("Unknown component type for identifier: {}", id);
+							break;
+					}
+				});
+			
 			log.trace("Fetch existing concept IDs...");
 			
-			final Set<String> missingConceptIds = fetchComponentIds(context, conceptIds, SnomedConceptDocument.class);
+			missingConceptIds.addAll(fetchComponentIds(context, conceptIds, SnomedConceptDocument.class));
+			final Function<String, Boolean> canBeSkipped = id -> {
+				return skippableMemberDependenciesByEffectiveTime.containsKey(id) && !dependenciesByEffectiveTime.containsKey(id);
+			};
 			
 			if (!missingConceptIds.isEmpty()) {
 				missingConceptIds.forEach(id -> {
-					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "concept");
+					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "concept", canBeSkipped.apply(id));
 				});
+				
 			}
 			
 			log.trace("Fetch existing description IDs...");
 			
-			final Set<String> missingDescriptionIds = fetchComponentIds(context, descriptionIds, SnomedDescriptionIndexEntry.class);
+			missingDescriptionIds.addAll(fetchComponentIds(context, descriptionIds, SnomedDescriptionIndexEntry.class));
 
 			if (!missingDescriptionIds.isEmpty()) {
 				missingDescriptionIds.forEach(id -> {
-					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "description");
+					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "description", canBeSkipped.apply(id));
 				});
 			}
 			
 			log.trace("Fetch existing relationship IDs...");
 
-			final Set<String> missingRelationshipIds = fetchComponentIds(context, relationshipIds, SnomedRelationshipIndexEntry.class);
+			missingRelationshipIds.addAll(fetchComponentIds(context, relationshipIds, SnomedRelationshipIndexEntry.class));
 
 			if (!missingRelationshipIds.isEmpty()) {
 				missingRelationshipIds.forEach(id -> {
-					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "relationship");
+					reportMissingComponent(globalDefectAcceptor, id, dependenciesByEffectiveTime.get(id), "relationship", canBeSkipped.apply(id));
 				});
 			}
-			
+		}
+		
+		removeSkippableMembers(skippableMemberDependenciesByEffectiveTime, dependenciesByEffectiveTime, slices);
+	}
+	
+	private void removeSkippableMembers(Map<String, String> skippableMemberDependenciesByEffectiveTime, Map<String, String> dependenciesByEffectiveTime, List<Rf2EffectiveTimeSlice> slices) {
+		
+		for (Rf2EffectiveTimeSlice slice : slices) {
```

Comment:
Could you please run this on a few RF2 inputs and measure the performance? Thank you!
Also, make sure you run this on the UK CL/DR/PA RF2 releases, they often have issues with reference set members pointing to non-existed components.